stdlib/socket.pyi

```diff
@@ -538,6 +538,16 @@ AI_V4MAPPED_CFG: AddressInfo
 if sys.platform == "win32":
     errorTab: dict[int, str]  # undocumented
 
+class _SendableFile(Protocol):
+    def read(self, __size: int) -> bytes: ...
+
+    # optional fields:
+    #
+    # @property
+    # def mode(self) -> str: ...
+    # def fileno(self) -> int: ...
+    # def seek(self, __offset: int) -> object: ...
```

Comment:
If offset is given, `_sendfile_use_send` calls this method unconditionally (at least on main). We should perhaps put it in the protocol.